stellar-contract-env-common/src/env_val.rs

```diff
@@ -123,6 +123,25 @@ pub trait TryIntoEnvVal<E: Env, V>: Sized {
     fn try_into_env_val(self, env: &E) -> Result<EnvVal<E, V>, Self::Error>;
 }
 
+impl<E: Env, F, T> TryIntoEnvVal<E, T> for F
+where
+    EnvVal<E, F>: TryInto<T>,
+{
+    type Error = <EnvVal<E, F> as TryInto<T>>::Error;
+
+    fn try_into_env_val(self, env: &E) -> Result<EnvVal<E, T>, Self::Error> {
+        let ab: T = EnvVal {
+            env: env.clone(),
+            val: self,
+        }
+        .try_into()?;
+        Ok(EnvVal {
+            env: env.clone(),
+            val: ab,
+        })
```

Comment:
All these EnvVal constructions look a bit intimidating from a performance point-of-view, especially since the latter one will be discarded in most places when used with `TryFromVal::try_from_val`, but it doesn't really matter because the for the wasm builds the `EnvVal` struct will have a zero sized `env`.